crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -41,13 +46,17 @@ reveal_type(c_instance.declared_and_bound)  # revealed: bool
 # mypy and pyright do not show an error here.
 reveal_type(c_instance.possibly_undeclared_unbound)  # revealed: str
 
+reveal_type(c_instance.inferred_from_redefined_param)  # revealed: Unknown | None | int
+
+reveal_type(c_instance.inferred_from_param_not_in_init)  # revealed: Unknown | str
```

Comment:
This test case also makes sense, but we already have that below in the "Variable defined in non-`__init__` method" section. Notice that it comes with a TODO that shows that we want to infer the declared parameter type for these cases as well. I think it would be surprising to special-case `__init__` in the way you have done here in this PR, although I understand the intention.

So to summarize: I think we can remove this test case here; remove the `if name.as_str() == "__init__"` check in your implementation, and then resolve the TODO in the test case below.